redwood-protocol-host/src/commonMain/kotlin/app/cash/redwood/protocol/host/HostProtocolAdapter.kt

```diff
@@ -57,10 +60,12 @@ public class HostProtocolAdapter<W : Any>(
     is GeneratedProtocolFactory -> factory
   }
 
-  private val nodes = mutableMapOf<Id, ProtocolNode<W>>(
-    Id.Root to RootProtocolNode(container),
-  )
-  private val changedWidgets = mutableSetOf<ChangeListener>()
+  private val nodes =
+    mutableIntObjectMapOf<ProtocolNode<W>>(Id.Root.value, RootProtocolNode(container))
+
+  private val removeNodeById: (Id) -> Unit = { nodes.remove(it.value) }
```

Comment:
This lambda causes boxing and will need switched to a dedicated `fun interface` in a follow-up.